File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/network/model/response/GetUserResponse.java
package com.lattis.ellipse.data.network.model.response;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class GetUserResponse extends AbstractApiResponse {
    @SerializedName("payload")
    GetUserPayloadResponse getUserPayloadResponse;

    public UserResponse getUserResponse() {
        return getUserPayloadResponse.getUserResponse();
    }

    public List<PrivateNetworkResponse> getPrivateNetworkResponse() {
        return getUserPayloadResponse.getPrivateNetworkResponse();
    }

    @Override
    public String toString() {
        return "GetUserResponse{" +
                "userResponse=" + getUserPayloadResponse +
                '}';
    }
}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/entity/response/user/GetUserResponse.kt
package com.lattis.data.entity.response.user

import com.google.gson.annotations.SerializedName
import com.lattis.data.entity.response.base.AbstractApiResponse
import com.lattis.data.entity.response.private_networks.PrivateNetworkResponse
import com.lattis.data.entity.response.user.UserResponse
import com.lattis.domain.models.User

class GetUserResponse : AbstractApiResponse() {
    @SerializedName("payload")
    var getUserPayloadResponse: GetUserPayloadResponse? = null

    fun getUserPayload():UserResponse?{
        return getUserPayloadResponse?.userResponse
    }

    var userResponse: UserResponse? = getUserPayloadResponse?.userResponse

    val privateNetworkResponse: List<PrivateNetworkResponse>?
        get() = getUserPayloadResponse?.privateNetworkResponse

    override fun toString(): String {
        return "GetUserResponse{" +
                "userResponse=" + getUserPayloadResponse +
                '}'.toString()
    }
}
